wagtail_live/static/wagtail_live/js/intervalpolling.js

```diff
@@ -0,0 +1,201 @@
+
+                /** Update the timestamp of the last update received. */
+                lastUpdateReceivedAt = lastUpdateTS["last_update_timestamp"];
+                
+                setTimeout(async () => await getUpdates(), POLLING_INTERVAL);
+            }
+
+        } else {
+            /** No updates are available, wait for the polling interval duration and call getUpdates. */
+            setTimeout(async () => await getUpdates(), POLLING_INTERVAL)
+        };
+    }
+}
+
+/**
+ * Fetches timestamp of the last update for this page.
+ * @returns {*} HttpResponse with headers containing the
+ *  timestamp of the last update for this page.
+ */
+async function fetchLastUpdateAt() {
+    return await fetch(basePollingURL, {method: 'HEAD'});
+}
+
+/**
+ * Checks if new updates are available.
+ * We know that new updates are available if the timestamp of
+ * the last update of the current page is greater than the
+ * timestamp of the last update received in the client side.
+ * @param {*} response - HttpResponse from HEAD request 
+ * @returns {boolean} true if new updates are available, false else.
+ */
+function newUpdate(response) {
+    let tsLastUpdateAt = response.headers.get('Last-Update-At');
+    if (parseFloat(tsLastUpdateAt) > parseFloat(lastUpdateReceivedAt)) {
+        return true;
+    } else {return false;}
+}
+
+/**
+ * Fetches new updates from the server by sending the timestamp
+ * of the last update received in the client side. 
+ * Server should respond with new updates, current live posts and 
+ * the timestamp of the page's last update.
+ * @returns {*} HttpResponse
+ */
+async function fetchUpdates() {
+    let url = basePollingURL + '?' +  new URLSearchParams({last_update_ts: lastUpdateReceivedAt});
+    return await fetch(url);
+}
+
+/**
+ * Retrievs a live post by its ID.
+ * @param {str} livePostID - Live post's ID
+ * @returns {HTMLElement} corresponding to the live post if it exists, else null.
+ */
+function getPostByID(livePostID) {
+    return document.querySelector(`[data-post-id='${livePostID}']`);
+}
+
+/**
+ * Processes new updates.
+ * Replaces a previous live post if it's been edited or 
+ * adds a new live post if it's been created.
+ * @param {string} updateID - ID of the live post to update
+ * @param {string} value - Value of the new content of this live post.
+ */
+function process(updateID, value) {
+    let postsDiv = document.querySelector("#live-posts");
+    let livePost = createLivePostWrapper(value);
+    let post = getPostByID(updateID);
+
+    if (post != null) {
+        postsDiv.replaceChild(livePost, post.parentElement);
+    } else {postsDiv.insertAdjacentElement("afterbegin", livePost);}
+}
+
+/**
+ * Removes a live post.
+ * @param {string} livePostID - ID of the live post to remove.
+ */
+function removeLivePost(livePostID) {
+    let post = getPostByID(livePostID);
+    if (post == null) {
+        /** Problematic! */
```

Comment:
Or not, since apparently it's already gone :)

Reply:
Alright!